src/Facets/GenericSwapFacet.sol

```diff
@@ -3,42 +3,328 @@ pragma solidity 0.8.17;
+    /// @param _integrator the name of the integrator
+    /// @param _referrer the address of the referrer
+    /// @param _receiver the address to receive the swapped tokens into (also excess tokens)
+    /// @param _minAmountOut the minimum amount of the final asset to receive
+    /// @param _swapData an object containing swap related data to perform swaps before bridging
+    function swapTokensSingleNativeToERC20(
+        bytes32 _transactionId,
+        string calldata _integrator,
+        string calldata _referrer,
+        address payable _receiver,
+        uint256 _minAmountOut,
+        LibSwap.SwapData calldata _swapData
+    ) external payable {
+        // ensure that contract (callTo) and function selector are whitelisted
+        if (
+            !(LibAllowList.contractIsAllowed(_swapData.callTo) &&
+                LibAllowList.selectorIsAllowed(bytes4(_swapData.callData[:4])))
+        ) revert ContractCallNotAllowed();
+
+        // get initial token balance
+        uint256 initialBalance = IERC20(_swapData.receivingAssetId).balanceOf(
+            address(this)
+        );
+
+        // execute swap
+        // solhint-disable-next-line avoid-low-level-calls
+        (bool success, bytes memory res) = _swapData.callTo.call{
+            value: msg.value
+        }(_swapData.callData);
+        if (!success) {
+            string memory reason = LibUtil.getRevertMsg(res);
+            revert(reason);
+        }
+
+        // get final balance
+        uint256 amountReceived = IERC20(_swapData.receivingAssetId).balanceOf(
+            address(this)
+        ) - initialBalance;
+
+        // ensure that minAmountOut was received
+        if (amountReceived < _minAmountOut)
+            revert CumulativeSlippageTooHigh(_minAmountOut, amountReceived);
+
+        // transfer funds to receiver
+        IERC20(_swapData.receivingAssetId).safeTransfer(
+            _receiver,
+            amountReceived
+        );
+
+        // emit events (both required for tracking)
+        emit AssetSwapped(
+            _transactionId,
+            _swapData.callTo,
+            _swapData.sendingAssetId,
+            _swapData.receivingAssetId,
+            _swapData.fromAmount,
+            _minAmountOut,
+            block.timestamp
+        );
+
+        emit LiFiGenericSwapCompleted(
+            _transactionId,
+            _integrator,
+            _referrer,
+            _receiver,
+            _swapData.sendingAssetId,
+            _swapData.receivingAssetId,
+            _swapData.fromAmount,
+            amountReceived
+        );
+    }
+
+    //----------------------------------------------------------------------------------------------------------------
+    //----------------------------------------------------------------------------------------------------------------
+    //----------------------------------------------------------------------------------------------------------------
```

Comment:
What are those lines for?

Reply:
They were just for me, differentiating between v1 and v2. 
Removed